talk/concurrency/condition.tex

```diff
@@ -48,43 +48,51 @@
 \end{frame}
 
 \begin{frame}[fragile]
+  \frametitlecpp[11]{Using condition variables: wait}
+  \vspace{-1.2\baselineskip}
   \begin{overprint}
   \onslide<1>
-  \begin{block}{Consumer side I: Going into wait}
+  \begin{block}{Mechanics of wait}
     \begin{itemize}
-      \item Start many threads which have to wait for shared data
-      \item Provide a lock to be managed by \mintinline{cpp}{wait}
-      \item \mintinline{cpp}{wait} will only lock while necessary; unlocked while sleeping
-      \item Threads might wake up, but \mintinline{cpp}{wait} returns only when condition satisfied
+      \item Many threads are waiting for shared data
+      \item Pass a \mintinline{cpp}{unique_lock} and a predicate for wakeup to \mintinline{cpp}{wait()}
+      \item \mintinline{cpp}{wait()} sends threads to sleep while predicate is \mintinline{cpp}{false}
+      \item \mintinline{cpp}{wait()} will only lock while necessary; unlocked while sleeping
+      \item Threads might wake up, but \mintinline{cpp}{wait()} returns only when lock available \emph{and} predicate \mintinline{cpp}{true}
```

Comment:
Should we mention the term `spurious wakeup` here?

Reply:
Orally on the lecture? I can add it to the slides as well if you prefer.

Reply:
I meant mentioning on the slides in written form :)